### PR TITLE
remove unnecessary FT for type restriction

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

This is needed for autodiff to work.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/cache/precomputed_quantities.jl
Comment (szy21) [inline src/cache/precomputed_quantities.jl:1]: @charleskawczynski @dennisYatunin Could you take a look at whether this is a preferred pattern? If so I will go ahead and change all the functions in the code.

Comment (charleskawczynski) [reply src/cache/precomputed_quantities.jl:1]: This is probably fine. We could even remove `::FT` in `z::FT` (and `where {FT}`) and replace `FT(0)` with `zero(z)`. The main risk with this is that if we do accidentally introduce type instabilities, this function could majorly slow down-- and diagonal dispatch will protect us from happening. But if the syntax is too burdensome, then I'm fine with trying to remove it.

File: src/utils/utilities.jl [hunk 1]
```diff
@@ -25,7 +25,7 @@ has_no_source_or_sink(parsed_args) = all((
 
 # we may be hitting a slow path:
 # https://stackoverflow.com/questions/14687665/very-slow-stdpow-for-bases-very-close-to-1
-fast_pow(x::FT, y::FT) where {FT <: AbstractFloat} = exp(y * log(x))
+fast_pow(x::FT, y) where {FT <: AbstractFloat} = exp(y * log(x))
```
Comment (charleskawczynski) [inline src/utils/utilities.jl:28]:
```suggestion
fast_pow(x, y) = exp(y * log(x))
```
We can remote `FT` entirely if it's not used

**Comment by szy21** [comment]
@charleskawczynski Do you have more comments on this PR?

**Review by charleskawczynski — APPROVED** [review]
No, I think the changes are fine

**Pull request merged**